ingest/src/test/java/com/microsoft/azure/kusto/ingest/CloseableByteArrayInputStream.java

```diff
@@ -0,0 +1,28 @@
+package com.microsoft.azure.kusto.ingest;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+
+public class CloseableByteArrayInputStream extends java.io.ByteArrayInputStream {
+    private boolean closed;
+
+    public CloseableByteArrayInputStream(byte[] buf) {
+        super(buf);
+
+        closed = false;
+    }
+
+    @Override
+    public int read(byte[] b) throws IOException {
+        if (closed) {
+            throw new IOException("Stream is closed");
+        }
+        return super.read(b);
+    }
+
+    @Override
+    public void close() {
+        closed = true;
```

Comment:
Since we don't call `super()`, it looks like it's not actually closed. I need to reread this PR to better understand the semantics and intent here, but either way I suspect we can name the class to be clearer that we don't allow the stream to be closed the traditional way.

Reply:
ByteArrayInputStream's close is a no-op:
https://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayInputStream.html

Which is why I made this class.
I could call super for clarity, but it doesn't do anything.

Reply:
Nah, no need. I think just a comment would help.
Thanks for explaining.

Reply:
Is the comment ok?

Reply:
Great, thanks! See nit comment, though